### PR TITLE
[#IOCOM-987] Invalidate clickable links in email

### PR DESCRIPTION
**Pull request by Garma00**
#### List of Changes

* [Replace links with a warning](https://github.com/pagopa/io-functions-services/commit/eaad9b401b2cd2928d3d462a111831d3b8e56fe2);
* [Invalidate clickable links](https://github.com/pagopa/io-functions-services/commit/c17647a887e0864be89a936397ff2bb746af22cc);

The text where we are using the regex will always been of 134 character so we are not affected by the problem highlighted by sonar.

#### Motivation and Context

We want to avoid clickable links in email

#### How Has This Been Tested?

Unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: EmailNotification/__tests__/utils.test.ts [hunk 1]
```diff
@@ -78,18 +79,48 @@ describe("removeLinks", () => {
...
-    expect(removeLinks(`${baseText}${simpleLink}`)).toBe(baseText);
+    expect(removeLinks(`${baseText} ${simpleLink}`)).toBe(
+      `${baseText} [link rimosso]`
+    );
+  });
+});
+
+describe("invalidateClickableLinks", () => {
+  test("should return the same string if no period are provided", () => {
+    expect(invalidateClickableLinks("a simple text with no perios")).toBe(
```
Comment (michaeldisaro) [inline EmailNotification/__tests__/utils.test.ts:116]: typo `perios`

File: EmailNotification/__tests__/utils.test.ts [hunk 2]
```diff
@@ -78,18 +79,48 @@ describe("removeLinks", () => {
...
+      "a simple text with no perios"
```
Comment (michaeldisaro) [inline EmailNotification/__tests__/utils.test.ts:117]: ditto

File: EmailNotification/__tests__/utils.test.ts [hunk 3]
```diff
@@ -78,18 +79,48 @@ describe("removeLinks", () => {
...
+    );
+  });
+
+  test("should add a zero-width space before every '.' character", () => {
```
Comment (michaeldisaro) [inline EmailNotification/__tests__/utils.test.ts:121]: you say before, but it seems that it's after.

File: EmailNotification/utils.ts [hunk 4]
```diff
@@ -114,11 +114,24 @@ export const truncateMarkdown = (plainText: string): string =>
     : plainText.substring(0, MAX_CHARACTER_FOR_BODY_MAIL);
 
 export const removeLinks = (text: string): string =>
-  text.replace(/(?:https?|ftp):\/\/[\n\S]+/g, "");
+  text.replace(/(?:https?|ftp):\/\/[\n\S]+/g, "[link rimosso]");
```
Comment (michaeldisaro) [inline EmailNotification/utils.ts:117]: what if there's something like `xyz://anything.anything`? I'd suggest to find a regex that matches everything before a `://`

File: EmailNotification/__tests__/utils.test.ts (hunk 1 above)
Comment (Garma00) [reply EmailNotification/__tests__/utils.test.ts]: [22dbf96](https://github.com/pagopa/io-functions-services/pull/297/commits/22dbf96d45f66a64fd34a79160e0651cc02f0117)

File: EmailNotification/__tests__/utils.test.ts (hunk 3 above)
Comment (Garma00) [reply EmailNotification/__tests__/utils.test.ts]: [22dbf96](https://github.com/pagopa/io-functions-services/pull/297/commits/22dbf96d45f66a64fd34a79160e0651cc02f0117)

File: EmailNotification/utils.ts (hunk 4 above)
Comment (Garma00) [reply EmailNotification/utils.ts]: [935a00a](https://github.com/pagopa/io-functions-services/pull/297/commits/935a00aace0fb262f6a646020b13df7f9b2040fd)

**Review by michaeldisaro — APPROVED** [review]

**Pull request merged**